File: sample-conference-java/app/src/main/java/com/quickblox/sample/videochat/conference/java/fragments/ScreenShareFragment.java
package com.quickblox.sample.videochat.conference.java.fragments;

import android.content.Context;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CompoundButton;
import android.widget.ToggleButton;

import com.quickblox.sample.videochat.conference.java.R;
import com.quickblox.videochat.webrtc.QBRTCMediaConfig;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;


public class ScreenShareFragment extends BaseToolBarFragment {
    private static final String TAG = ScreenShareFragment.class.getSimpleName();

    private OnSharingEvents onSharingEvents;

    public static ScreenShareFragment newInstance() {
        return new ScreenShareFragment();
    }

    @Override
    int getFragmentLayout() {
        return R.layout.fragment_screensharing;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = super.onCreateView(inflater, container, savedInstanceState);

        if (view != null) {
            ToggleButton micToggle = view.findViewById(R.id.tb_switch_mic);
            ToggleButton cameraToggle = view.findViewById(R.id.tb_switch_cam);
            ToggleButton endCallToggle = view.findViewById(R.id.tb_end_call);
            ToggleButton shareScreenToggle = view.findViewById(R.id.tb_screen_share);
            ToggleButton swapCamToggle = view.findViewById(R.id.tb_swap_cam);

            micToggle.setVisibility(View.GONE);
            cameraToggle.setVisibility(View.GONE);
            endCallToggle.setVisibility(View.GONE);
            swapCamToggle.setVisibility(View.GONE);

            shareScreenToggle.setChecked(false);
            shareScreenToggle.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                @Override
                public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                    Log.d(TAG, "Stop Screen Sharing");
                    if (onSharingEvents != null) {
                        onSharingEvents.onStopPreview();
                    }
                }
            });
        }

        ImagesAdapter adapter = new ImagesAdapter(getChildFragmentManager());
        ViewPager pager = (ViewPager) view.findViewById(R.id.pager);
        pager.setAdapter(adapter);

        QBRTCMediaConfig.setVideoWidth(QBRTCMediaConfig.VideoQuality.HD_VIDEO.width);
        QBRTCMediaConfig.setVideoHeight(QBRTCMediaConfig.VideoQuality.HD_VIDEO.height);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            onSharingEvents = (OnSharingEvents) context;
        } catch (ClassCastException e) {
            throw new ClassCastException(getActivity().toString() + " must implement OnSharingEvents");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        onSharingEvents = null;
    }

    public static class ImagesAdapter extends FragmentPagerAdapter {
        private static final int NUM_ITEMS = 1;

        private int[] images = {R.drawable.omg_ican_see_img/*, R.drawable.p2p, R.drawable.group_call, R.drawable.opponents*/};

        ImagesAdapter(FragmentManager fm) {
            super(fm);
        }

        @Override
        public int getCount() {
            return NUM_ITEMS;
        }

        @Override
        public Fragment getItem(int position) {
            return PreviewFragment.newInstance(images[position]);
        }
    }

    public interface OnSharingEvents {

        void onStopPreview();
    }
}

File: old_samples/sample-videochat-kotlin/src/main/java/com/quickblox/sample/videochatkotlin/fragments/ScreenShareFragment.kt
package com.quickblox.sample.videochatkotlin.fragments

import android.content.Context
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.quickblox.sample.videochatkotlin.R
import kotlinx.android.synthetic.main.fragment_pager.*

class ScreenShareFragment : BaseToolBarFragment() {
    val TAG = ScreenShareFragment::class.java.simpleName
    private var onSharingEvents: OnSharingEvents? = null

    override val fragmentLayout: Int
        get() = R.layout.fragment_pager

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val adapter = MyAdapter(childFragmentManager)
        pager.adapter = adapter
    }

    override fun initActionBar() {
        actionBar.setBackgroundDrawable(ColorDrawable(ContextCompat.getColor(context!!, R.color.white)))
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        inflater!!.inflate(R.menu.screen_share_fragment, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        return when (item!!.itemId) {
            R.id.stop_screen_share -> {
                Log.d(TAG, "stop_screen_share")
                if (onSharingEvents != null) {
                    onSharingEvents!!.onStopSharingPreview()
                }
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        onSharingEvents = context as OnSharingEvents
    }

    override fun onDetach() {
        super.onDetach()
        onSharingEvents = null
    }

    interface OnSharingEvents {
        fun onStopSharingPreview()
    }

    companion object {
        fun newInstance() = ScreenShareFragment()
    }

    class MyAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

        private val images = intArrayOf(R.drawable.pres_img, R.drawable.splash_screen, R.drawable.users_screen)

        override fun getCount(): Int {
            return images.size
        }

        override fun getItem(position: Int): Fragment {
            return PreviewSharingFragment.newInstance(images[position])
        }
    }
}